### DISCUSSION TITLE
led_strip:  How to Set Value of W Channel of an RGBW Strip?

### DISCUSSION THREAD

**Discussion by Xylopyrographer** [UncleRus/esp-idf-lib#522] [Q&A] [answered]
If a LED strip is made up of RGBW LED's—SK6812 RGBW like [this](https://www.aliexpress.com/item/4000890862758.html?spm=a2g0o.productlist.main.17.3ed53a95nvpgAz&algo_pvid=c01e0e61-91e7-4d60-9c07-7874c1283781&algo_exp_id=c01e0e61-91e7-4d60-9c07-7874c1283781-8&pdp_npi=3%40dis%21CAD%2159.51%2136.3%21%21%21%21%21%4021227f0f16856881038095024d0745%2112000033375607146%21sea%21CA%214784559031&curPageLogUid=aIyBMB20mLTf) for example, how is the W channel value set?

The `rgb_t` structure only holds data for the R, G and B channels. If `is_rgbw` is set `true` in the `led_strip_t` descriptor, it seems that  `led_strip_set_pixel()` creates an `"rgb luma value"` for the W channel based on the RGB value for that LED position, but it doesn't  seem like there is a way to directly specify the W channel value independent of the RGB values?

**Comment by UncleRus** [comment, marked as answer]
Yes, you understand correctly: the white value for RGBW strips is calculated automatically and you cannot set it manually through the library interface. However, if you really need to, you can correct the white values yourself directly in the strip buffer. Something like this:
```C

led_strip_install();
...
led_strip_t strip = { ... };
...
led_strip_set_pixel(&strip, 0, rgb_from_values(0, 0, 100));
// here
strip.buf[3] = 255;
...
// display
led_strip_flush(&strip);
```

**Reply by Xylopyrographer** [reply]
Got it 😄!  Thanks for the reply.

**Comment by UncleRus** [comment]
BTW if you need, here is a more complex example of using this component:

https://github.com/UncleRus/EvLamp

**Reply by Xylopyrographer** [reply]
That is a significant piece of work! 👍
